src/irmin/tree.ml

```diff
@@ -973,9 +984,13 @@ module Make (P : Backend.S) = struct
           | `True, false -> (
               match t.v with
               | Map m -> (map [@tailcall]) ~path acc d (Some m) k
-              | Value (repo, _, _) | Key (repo, _) ->
+              | Value (repo, v, Some um) ->
+                  let t' = { t with v = Value (repo, v, None) } in
+                  let* v = to_value ~cache t' >|= get_ok "fold" in
```

Comment:
I don't understand these 2 lines. It seems to me that the `v` in `let* v` is the same as the one in `Value (repo, v, Some um) ->`.

Reply:
you are right :)